wikiscrape/export_pages.py

```diff
@@ -0,0 +1,101 @@
+"""Export the pages we enumerated as xml.
+
+This page https://www.mediawiki.org/wiki/Manual:Parameters_to_Special:Export
+lists multiple limits to the amount of data that can be returned. In the two
+wiki's I have been testing on I haven't found these to be true. The main
+points of concern are:
+* pages: The limit is 35
+* limit: The maximum number of revisions to return, limited at 1000.
+* history: It mentions there are cases where this doesn't return all the revisions.
+* listauthors: This didn't seem active on any of the wikis I tested on.
+"""
+
+
+import argparse
+import glob
+import os
+import urllib.parse
+from typing import List
+
+from utils import get_page, get_wiki_name
+
+parser = argparse.ArgumentParser(description="Export mediawikis as XML")
+parser.add_argument("--wiki", required=True, help="The wiki url we are exporting.")
+parser.add_argument(
+    "--pages",
+    action="append",
+    help="A list of files of pages to export, or a dir to export all. defaults to data/${wiki_name}/pages/.",
+)
+# Using firefox I didn't have issues sending a lot of pages at once, but I was
+# getting URI too long errors when using requests.
+parser.add_argument(
+    "--page_limit", default=35, help="The max number of pages to export at once."
+)
+parser.add_argument(
+    "--output_dir",
+    help="Where to save the xml export. defaults to data/${wiki_name}/export/.",
+)
+# TODO: Implement this if we find a wiki that has this enabled.
+parser.add_argument(
+    "--listauthors",
+    help="Use the listauthors url param instead of getting multiple revisions. UNIMPLEMENTED.",
```

Comment:
If you want to be fancy, raise a NotImplementedError if this is set?

Reply:
Added